docs/articles_en/documentation/openvino-extensibility/transformation-api/patterns-python-api.rst

```diff
@@ -0,0 +1,130 @@
+.. {#openvino_docs_Extensibility_UG_patterns-python-api}
+
+Overview of creating pattern using Python API
+=============================================
+
+.. meta::
+   :description: Learn how to apply additional model optimizations or transform
+                 unsupported subgraphs and operations, using OpenVINO™ Transformations API.
+
+Pattern matching is an essential component of OpenVINO™ transformations. Before performing any transformation on a subgraph of a graph, we need to find the subgraph in the graph.
```

Comment:
I believe it's worth to mention Matcher pass page https://docs.openvino.ai/2024/documentation/openvino-extensibility/transformation-api/matcher-pass.html
As I can see, we already have an article about Patterns here
<img width="527" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/6aacb804-bf26-408a-82e9-e3eee80dceaa">

Should we delete it or combine them?

Reply:
I think it's better to add a ref to this pager from [matcher-pass.html](https://docs.openvino.ai/2024/documentation/openvino-extensibility/transformation-api/matcher-pass.html)
in this section 
<img width="536" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/1c63ff8f-d19f-4184-ba88-ae0411cb19f9">

Reply:
Let's add a link

Reply:
Added ref to the MatcherPass page. The MatcherPass page will require changes in the future.